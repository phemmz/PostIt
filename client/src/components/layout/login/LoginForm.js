import React, { Component } from 'react';
import classnames from 'classnames';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import GoogleLogin from 'react-google-login';
import sharedSigninValidations from '../../../../../server/shared/loginValidations';
import TextFieldGroup from '../../common/TextFieldGroup';
import AuthenticationActions from '../../../actions/authActions';
/**
 * LoginForm class
 */
class LoginForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            errors: {},
            isLoading: false
        }
/**
 * binds the onSubmit and onChange functions to this
 */
        this.onSubmit = this.onSubmit.bind(this);
        this.onChange = this.onChange.bind(this);
        this.responseGoogle = this.responseGoogle.bind(this);
    }
/**
 * isValid() gets called on login submit and sets any validation errors to the state
 */
    isValid() {
        const { errors, isValid } = sharedSigninValidations.validateSignin(this.state);
        if (!isValid) {
            this.setState({ errors });
        }
        return isValid; 
    }
/**
 * onSubmit() gets called when the login button is pressed, it calls isValid() to check
 * if there are no validation errors and then fires the login action
 */
    onSubmit(e) {
/**
 * prevents the default onSubmit action of a form tag
 */
        e.preventDefault();
        if (this.isValid()) {
          this.setState({ errors: {} });
          /**
           * the login action is fired here
           */
          this.props.login(this.state)
            .then((res) => {
              this.context.router.push('/dashboard');
              Materialize.toast('Login Successful!!', 4000, 'green');
            })
            .catch((err) => {
              Materialize.toast(err.response.data.errors.message, 4000, 'red')
              this.setState({ errors: err.response.data.errors})
            });
        }
    }
/**
 * onChange() gets called when the input field changes and the change is stored in the state
 */
    onChange(e) {
        this.setState({ [e.target.name]: e.target.value });
    }
    /**
     * Gets the userDetails as response from google
     * @param {*} response 
     */
    responseGoogle(response) {
      const userDetails = {
        username: response.profileObj.email.substring(0, response.profileObj.email.indexOf('@')),
        email: response.profileObj.email,
        password: response.profileObj.googleId,
        phoneNumber: response.profileObj.googleId
      }
      this.props.googleAuthentication(userDetails)
        .then((res) => {
          this.context.router.push('/dashboard');
          Materialize.toast('Welcome!!', 4000, 'green');
        })
    }

	render() {
/**
 * deconstruct variables from state
 */
    	const { errors, username, password, isLoading } = this.state;
    	return (
        <div className="container">
          <div className="row">
            <div className="col s12 m6 offset-s7 offset-m6 login">
              <div className="right">
                <GoogleLogin
                  clientId='484298663558-arrp0kt4u2j2aro9k2bbb2bcffth9fke.apps.googleusercontent.com'
                  onSuccess={this.responseGoogle}
                  onFailure={this.responseGoogle}
                  buttonText="Sign in with Google"
                  className="google-link"
                />
              </div>
              <form onSubmit={this.onSubmit}>
                <h4 className="green-text text-darken-4 login-text">Login</h4>	
                {/* renders error messages if there are error messages */}
                { errors.message  && <div id="uname-error" className="error">{errors.message}</div> }
                <TextFieldGroup 
                  error={errors.username}
                  id="username"
                  onChange={this.onChange}
                  value={this.state.username}
                  field="username"
                  htmlFor="username"
                  label="Username"
                />
                <TextFieldGroup 
                  error={errors.password}
                  id="password"
                  onChange={this.onChange}
                  value={this.state.password}
                  field="password"
                  htmlFor="password"
                  label="Password"
                  type="password"
                />
                <div className="center">
                  <button className="waves-effect waves-light btn">Login</button>
                </div>
                <div className="center reset-link">
                  <Link to="/reset">Forgot Password?</Link>
                </div>
              </form>	
            </div>
          </div>
        </div>
		)	
    }
}


const dispatchToProps = (dispatch) => {
  return {
    login: (data) => dispatch(AuthenticationActions.login(data)),
    googleAuthentication: (userDetails) => dispatch(AuthenticationActions.googleAuthentication(userDetails))
  }
}

LoginForm.contextTypes = {
    router: PropTypes.object.isRequired
}

export default connect(null, dispatchToProps)(LoginForm);