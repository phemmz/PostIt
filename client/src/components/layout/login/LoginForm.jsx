import React, { Component } from 'react';
import { Link, browserHistory } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import GoogleLogin from 'react-google-login';
import sharedSigninValidations from '../../../../../server/shared/loginValidations';
import TextFieldGroup from '../../common/TextFieldGroup.jsx';
import AuthenticationActions from '../../../actions/authActions';
/**
 * LoginForm class
 */
class LoginForm extends Component {
  /**
   * constructor
   * @param {*} props
   * @return {*} void
   */
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      errors: {},
      isLoading: false
    };
/**
 * binds the onSubmit and onChange functions to this
 */
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.responseGoogle = this.responseGoogle.bind(this);
  }

/**
 * onSubmit() gets called when the login button is pressed, it calls isValid() to check
 * if there are no validation errors and then fires the login action
 * @param {*} event
 * @return {*} void
 */
  onSubmit(event) {
/**
 * prevents the default onSubmit action of a form tag
 */
    event.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      /**
       * the login action is fired here
       */
      this.props.login(this.state)
        .then(() => {
          browserHistory.push('/dashboard');
          Materialize.toast('Login Successful!!', 4000, 'green');
        })
        .catch((err) => {
          Materialize.toast(err.response.data.errors.message, 4000, 'red');
          this.setState({ errors: err.response.data.errors });
        });
    }
  }
/**
 * onChange() gets called when the input field changes and the change is stored in the state
 * @param {*} event
 * @return {*} void
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
/**
 * Gets the userDetails as response from google
 * @param {*} response
 * @return {*} void
 */
  responseGoogle(response) {
    const userDetails = {
      username: response.profileObj.email.substring(0, response.profileObj.email.indexOf('@')),
      email: response.profileObj.email,
      password: response.profileObj.googleId,
      phoneNumber: response.profileObj.googleId
    };
    this.props.googleAuthentication(userDetails)
      .then(() => {
        browserHistory.push('/dashboard');
        Materialize.toast('Welcome!!', 4000, 'green');
      });
  }
/**
 * isValid() gets called on login submit and sets any validation errors to the state
 * @return {*} isValid
 */
  isValid() {
    const { errors, isValid } = sharedSigninValidations.validateSignin(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
/**
 * render
 * @return {*} void
 */
  render() {
/**
 * deconstruct variables from state
 */
    const { errors } = this.state;
    return (
      <div className="container">
        <div className="row">
          <div className="col s12 m6 offset-s7 offset-m6 login">
            <div className="right">
              <GoogleLogin
                clientId="484298663558-arrp0kt4u2j2aro9k2bbb2bcffth9fke.apps.googleusercontent.com"
                onSuccess={this.responseGoogle}
                onFailure={this.responseGoogle}
                buttonText="Sign in with Google"
                className="google-link"
              />
            </div>
            <form onSubmit={this.onSubmit}>
              <h4 className="green-text text-darken-4 login-text">Login</h4>
              { errors.message && <div id="uname-error" className="error">{ errors.message }</div> }
              <TextFieldGroup
                error={errors.username}
                id="username"
                onChange={this.onChange}
                value={this.state.username}
                field="username"
                htmlFor="username"
                label="Username"
              />
              <TextFieldGroup
                error={errors.password}
                id="password"
                onChange={this.onChange}
                value={this.state.password}
                field="password"
                htmlFor="password"
                label="Password"
                type="password"
              />
              <div className="center">
                <button className="waves-effect waves-light btn">Login</button>
              </div>
              <div className="center reset-link">
                <Link to="/reset">Forgot Password?</Link>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

LoginForm.propTypes = {
  login: PropTypes.func.isRequired,
  googleAuthentication: PropTypes.func.isRequired
};

const dispatchToProps = (dispatch) => {
  return {
    login: (data) => {
      return dispatch(AuthenticationActions.login(data));
    },
    googleAuthentication: (userDetails) => {
      return dispatch(AuthenticationActions.googleAuthentication(userDetails));
    }
  };
};

export default connect(null, dispatchToProps)(LoginForm);
