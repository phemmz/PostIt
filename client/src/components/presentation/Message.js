import React, { Component } from 'react';

class Message extends Component {
	render() {
		return (
			<div className="email-content-wrap">
				<div className="row">
					<div className="col s12 m10 l10 msg-position">
						<ul className="collection">
							<li className="collection-item avatar">
								<span className="circle green lighten-1 center">{this.props.currentMessage.messagecreator[0]}</span>
								<span className="email-title">{this.props.currentMessage.messagecreator}</span>
								<p className="grey-text ultra-small">Priority: {this.props.currentMessage.priority}</p>
								<p className="grey-text ultra-small">{new Date(this.props.currentMessage.createdAt).toLocaleString()}</p>
								<span className="email-content">{this.props.currentMessage.content}</span>
							</li>
						</ul>
					</div>
				</div>
				<hr />
			</div>
		)
	}
}
export default Message;