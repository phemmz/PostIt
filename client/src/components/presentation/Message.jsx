import React from 'react';
import PropTypes from 'prop-types';

const Message = ({
  currentMessage
}) => {
  return (
    <div className="email-content-wrap">
      <div className="row">
        <div className="col s12 m10 l10 msg-position">
          <ul className="collection">
            <li className="collection-item avatar">
              <span
                className="circle green lighten-1 center"
              >{currentMessage.messagecreator[0]}</span>
              <span
                className="email-title"
                id="msg-creator"
              >{currentMessage.messagecreator}</span>
              <span
                className="grey-text ultra-small right date-msg-position"
              >Priority: {currentMessage.priority}</span>
              <span
                className="grey-text ultra-small right date-pad date-msg-position"
              >{new Date(currentMessage.createdAt).toLocaleString()}</span>
              <div className="flex wordwrap">
                <span
                  className="email-content msg-content"
                >{currentMessage.content}</span>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <hr />
    </div>
  );
};

Message.propTypes = {
  currentMessage: PropTypes.object.isRequired,
};

export default Message;
