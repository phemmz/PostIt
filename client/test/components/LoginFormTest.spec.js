import React from 'react';
import { mount, shallow } from 'enzyme';
import LoginForm from '../../src/components/layout/login/LoginForm';
import Message from '../../src/components/presentation/Message';

describe('LoginForm', () => {
  it('renders form and h1', () => {
    const wrapper = shallow(<h1>Hello</h1>);
    expect(wrapper).toMatchSnapshot();
    // const wrapper = setup();
    // expect(wrapper.find('form')).toExist;
    // expect(wrapper.find('Row')).toExist;
    // expect(wrapper.find('h1').text()).toEqual('Login');
    // expect(wrapper.find('div')).toExist;
  });
  it('should render Message', () => {
    const message = {
      content: 'hello man',
      createdAt: '2017-08-03T12:33:32.666Z',
      groupId: 1,
      id: 1,
      messagecreator: 'phemmz',
      priority: 1,
      readcheck: null,
      updatedAt: '2017-08-03T12:33:32.666Z',
      userId: 1,
    }
    const wrapper = shallow(
      <Message currentMessage={message} />
    );
    expect(wrapper.find('strong')).toExist;
  });
});
