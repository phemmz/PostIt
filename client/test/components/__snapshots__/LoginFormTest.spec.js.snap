// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders form and h1 1`] = `
<h1>
  Hello
</h1>
`;
